lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -298,6 +297,31 @@ public void onAuthUpdated() {
 		}
 	}
 
+	/**
+	 * Called when where was an error during authentication attempt
+	 *
+	 * @param errorInfo Error associated with unsuccessful authentication
+	 */
+	public void onAuthError(ErrorInfo errorInfo) {
+		Log.i(TAG, String.format("onAuthError: (%d) %s", errorInfo.code, errorInfo.message));
+		switch (state.state) {
+			case connecting:
+				ITransport transport = this.transport;
+				if (transport != null)
+					/* onTransportUnavailable will send state change event and set transport to null */
+					onTransportUnavailable(transport, null, errorInfo);
```

Comment:
The error can happen earlier that token is transmitted to the server and server has a chance to reject it. If authUrl lookup fails or authCallback throws an exception for example